### ISSUE TITLE
When using BigNumber, toTex cannot display a large number well.

### ISSUE THREAD

**Issue by hadongzhu** [josdejong/mathjs#3163]
**Describe the bug**
When using BigNumber, toTex displays a large number as infinity, but it is normal when using toString.

This may be a problem caused by the toTex function of ConstantNode. toTex uses the built-in isFinite to determine whether the BigNumber is infinite, but it will make a misjudgment for a very large BigNumber. It seems that using its own isFinite for BigNumber would be better.

https://github.com/josdejong/mathjs/blob/4c42a14c8154a408f6fddbfdda9dcefde0818a9e/src/expression/node/ConstantNode.js#L136-L170

For example:
```javascript
math.config({number: "BigNumber"})
isFinite(math.evaluate("2^10000000")) // false
math.evaluate("2^10000000").isFinite() // true
```

**To Reproduce**
```javascript
math.config({number: "BigNumber"})
math.parse(math.evaluate("2^10000000").toString()).toString() // '9.049817306360800301396402667708707356248762388018002675910921592e+3010299'
math.parse(math.evaluate("2^10000000").toString()).toTex() // '\\infty'

**Comment by josdejong** [comment]
Thanks for reporting! Using `isFinite` for BigNumbers indeed doesn't work. Fixed via 50b15a751d61f3fedabe0078b919af2169aeb59a (not yet published).

**Comment by josdejong** [comment]
Fixed now in `v12.4.1`
